test/extensions/filters/udp/dns_filter/dns_filter_integration_test.cc

```diff
@@ -217,7 +216,7 @@ name: listener_1
   NiceMock<Stats::MockCounter> mock_record_name_overflow_;
   NiceMock<Stats::MockCounter> query_parsing_failure_;
   DnsParserCounters counters_;
-  std::unique_ptr<DnsMessageParser> response_parser_;
+  std::unique_ptr<Utils::DnsResponseValidator> response_parser_;
```

Comment:
You will not need this object if all the class methods are made static. Also, regardless, it doesn't need to be wrapped in a unique_ptr, as we're just calling the default constructor, and instantiation is not conditional.